File: 프로그래머스/Level2/올바른 괄호.py
# https://programmers.co.kr/learn/courses/30/lessons/12909

def solution(s):
    stack = []
    for i in s:
        if len(stack) == 0:
            stack.append(i)
            continue
        if stack[-1] == '(' and i == ')':
            stack.pop()
            continue
        stack.append(i)
    if len(stack) == 0:
        return True
    return False

File: 프로그래머스/Level2/올바른 괄호.js
// https://programmers.co.kr/learn/courses/30/lessons/12909

const solution = (s) => {
	const stack = [];
	for (let i = 0; i < s.length; i++) {
		if (stack.length === 0 && s[i] === ')') return false;
		if (s[i] === ')') {
			const temp = stack.pop();
			if (temp === ')') return false;
		} else stack.push(s[i]);
	}
	return stack.length === 0 ? true : false;
};
